tests/e2e/c/interchain_workflow.go

```diff
@@ -0,0 +1,160 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package c
+
+import (
+	"math/big"
+
+	ginkgo "github.com/onsi/ginkgo/v2"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/coreth/core/types"
+	"github.com/ava-labs/coreth/plugin/evm"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/tests/e2e"
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/utils/crypto/secp256k1"
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/ava-labs/avalanchego/utils/units"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+	"github.com/ava-labs/avalanchego/wallet/subnet/primary/common"
+)
+
+var _ = e2e.DescribeCChain("[Interchain Workflow]", func() {
+	require := require.New(ginkgo.GinkgoT())
+
+	const (
+		txAmount = 10 * units.Avax // Arbitrary amount to send and transfer
+		gasLimit = uint64(21000)   // Standard gas limit
+	)
+
+	ginkgo.It("should ensure that funds can be transferred from the C-Chain to the X-Chain and the P-Chain", func() {
+		ginkgo.By("allocating a pre-funded key to send from and a recipient key to deliver to")
+		senderKey := e2e.Env.AllocateFundedKey()
+		senderEthAddress := evm.GetEthAddress(senderKey)
+		factory := secp256k1.Factory{}
+		recipientKey, err := factory.NewPrivateKey()
+		require.NoError(err)
+		recipientEthAddress := evm.GetEthAddress(recipientKey)
+
+		ethClient := e2e.Env.NewEthClient()
+
+		ginkgo.By("sending funds from one address to another on the C-Chain", func() {
+			// Create transaction
+			acceptedNonce, err := ethClient.AcceptedNonceAt(e2e.DefaultContext(), senderEthAddress)
+			require.NoError(err)
+			gasPrice, err := ethClient.SuggestGasPrice(e2e.DefaultContext())
+			require.NoError(err)
+			tx := types.NewTransaction(
+				acceptedNonce,
+				recipientEthAddress,
+				big.NewInt(int64(txAmount)),
+				gasLimit,
+				gasPrice,
+				nil,
+			)
+
+			// Sign transaction
+			cChainID, err := ethClient.ChainID(e2e.DefaultContext())
+			require.NoError(err)
+			signer := types.NewEIP155Signer(cChainID)
+			signedTx, err := types.SignTx(tx, signer, senderKey.ToECDSA())
+			require.NoError(err)
+
+			require.NoError(ethClient.SendTransaction(e2e.DefaultContext(), signedTx))
+
+			ginkgo.By("waiting for the C-Chain recipient address to have received the sent funds")
+			e2e.Eventually(func() bool {
+				balance, err := ethClient.BalanceAt(e2e.DefaultContext(), recipientEthAddress, nil)
+				require.NoError(err)
+				return balance.Cmp(big.NewInt(0)) > 0
+			}, e2e.DefaultTimeout, e2e.DefaultPollingInterval, "failed to see funds delivered before timeout")
+		})
+
+		// Wallet must be initialized after sending funds on the
+		// C-Chain to ensure wallet state matches on-chain state
```

Comment:
This is technically still racy as is. In the above code we wait until a random node has accepted the C-chain tx. In the below code we assume a, potentially different, random node has accepted the C-chain tx. Which isn't guaranteed to be true. If we use the same URI - this should always work as expected.